hi-ml-histopathology/testhisto/testhisto/utils/utils_slides_datamodule.py

```diff
@@ -0,0 +1,248 @@
+import os
+from pathlib import Path
+import py
+import medmnist
+import numpy as np
+import pandas as pd
+import torchvision.transforms as transforms
+
+from enum import Enum
+from torch import Tensor
+from medmnist import INFO
+from tifffile import TiffWriter
+
+from torch.utils.data import DataLoader
+from typing import Tuple, Optional, List, Union
+from histopathology.datamodules.base_module import SlidesDataModule
+from histopathology.datasets.base_dataset import SlidesDataset
+from health_azure.utils import PathOrString
+
+
+class MockSlidesDataset(SlidesDataset):
+    """Mock and child class of SlidesDataset, to be used for testing purposes.
+    It overrides the following, according to the PANDA cohort settings:
+
+    :param SLIDE_ID_COLUMN: CSV column name for slide ID set to "image_id".
+    :param LABEL_COLUMN: CSV column name for tile label set to "isup_grade".
+    :param METADATA_COLUMNS: Column names for all the metadata available on the CSV dataset file.
+    """
+
+    SLIDE_ID_COLUMN = "image_id"
+    LABEL_COLUMN = "isup_grade"
+    METADATA_COLUMNS = ("data_provider", "isup_grade", "gleason_score")
+
+    def __init__(
+        self, root: PathOrString, dataset_csv: Optional[PathOrString] = None, dataset_df: Optional[pd.DataFrame] = None
+    ) -> None:
+        """
+        :param root: Root directory of the dataset.
+        :param dataset_csv: Full path to a dataset CSV file, containing at least
+        `TILE_ID_COLUMN`, `SLIDE_ID_COLUMN`, and `IMAGE_COLUMN`. If omitted, the CSV will be read
+        from `"{root}/{DEFAULT_CSV_FILENAME}"`.
+        :param dataset_df: A potentially pre-processed dataframe in the same format as would be read
+        from the dataset CSV file, e.g. after some filtering. If given, overrides `dataset_csv`.
+        """
+        super().__init__(root, dataset_csv, dataset_df, validate_columns=False)
+        slide_ids = self.dataset_df.index
+        self.dataset_df[self.IMAGE_COLUMN] = slide_ids + ".tiff"
+        self.validate_columns()
+
+
+class MockSlidesDataModule(SlidesDataModule):
+    """Mock and child class of SlidesDataModule, overrides get_splits so that it uses MockSlidesDataset."""
+
+    def get_splits(self) -> Tuple[MockSlidesDataset, MockSlidesDataset, MockSlidesDataset]:
+        return (MockSlidesDataset(self.root_path), MockSlidesDataset(self.root_path), MockSlidesDataset(self.root_path))
+
+
+class MockWSIType(Enum):
+    PATHMNIST = "pathmnist"
+    FAKE = "fake"
+
+
+class MockWSIGenerator:
+    """Generator class to create mock WSI on the fly. A mock WSI resembles to:
+                                [**      ]
+                                [  **    ]
+                                [    **  ]
+                                [      **]
+        where * reprents 2 tiles stitched along the Y axis.
+    :param METADATA_POSSIBLE_VALUES: Possible values to be assigned to the dataset metadata.
```

Comment:
For documentation to render correctly in sphinx, there needs to be a line between the description and the start of the :param section